src/routes/(app)/org/[id=integer]/club/[clubId]/attendance/+page.server.ts

```diff
@@ -0,0 +1,393 @@
+import { redirect } from '@sveltejs/kit';
+import crypto from 'crypto';
+import dayjs from 'dayjs';
+import { z } from 'zod';
+
+import { formHandler } from '$lib/bodyguard.js';
+import { getClubUserFromSession } from '$lib/server/getClubUserFromSession.js';
+import { prisma } from '$lib/server/prismaConnection.js';
+import { RRule } from '$lib/utils/rrule';
+
+export const load = async ({ parent, params, url }) => {
+	const parentData = await parent();
+	if (!parentData.clubPerms.viewAttendance) {
+		throw redirect(303, `/org/${params.id}/club/${params.clubId}`);
+	}
+
+	let event;
+
+	const eventIdString = url.searchParams.get('eventId');
+	if (eventIdString) {
+		event = await prisma.clubAttendanceEvent.findFirst({
+			where: {
+				AND: {
+					clubId: parentData.club.id,
+					id: parseInt(eventIdString)
+				}
+			}
+		});
+	} else {
+		event = await prisma.clubAttendanceEvent.findFirst({
+			where: {
+				clubId: parentData.club.id
+			},
+			orderBy: {
+				date: 'desc'
+			}
+		});
+	}
+
+	if (!event) {
+		event = await prisma.clubAttendanceEvent.create({
+			data: {
+				name: dayjs(Date.now()).format('MMMM DD, YYYY'),
+				clubId: parentData.club.id,
+				date: new Date()
+			}
+		});
+	}
+
+	const calEvents = await prisma.event.findMany({
+		where: {
+			clubId: parentData.club.id
+		}
+	});
+
+	const allEvents = await prisma.clubAttendanceEvent.findMany({
+		where: {
+			clubId: parentData.club.id
+		},
+		orderBy: {
+			date: 'desc'
+		}
+	});
+
+	const members = await prisma.clubUser.findMany({
+		where: {
+			clubId: parentData.club.id
+		},
+		select: {
+			user: {
+				select: {
+					firstName: true,
+					lastName: true,
+					pfp: true,
+					id: true,
+					attendanceMarks: {
+						where: {
+							clubId: parentData.club.id
+						},
+						select: {
+							attendanceEvent: true
+						}
+					}
+				}
+			}
+		},
+		orderBy: {
+			user: {
+				firstName: 'asc'
+			}
+		}
+	});
+
+	return {
+		attendanceMembers: members,
+		attendanceEvent: event,
+		allEvents,
+		calEvents
+	};
+};
+
+export const actions = {
+	create: formHandler(
+		z.object({
+			eventId: z.coerce.number(),
+			date: z.coerce.date().optional()
+		}),
+		async ({ eventId, date }, { cookies, params, url }) => {
+			const clubUser = await getClubUserFromSession(cookies.get('session'), params.clubId);
+
+			if (!clubUser.perms.admin && !clubUser.perms.manageAttendance) {
+				return {
+					success: false,
+					message: 'No permissions'
+				};
+			}
+
+			const event = await prisma.event.findFirst({
+				where: {
+					id: eventId,
+					clubId: clubUser.clubUser.clubId
+				}
+			});
+
+			if (!event) {
+				return {
+					success: false,
+					message: 'Invalid event id'
+				};
+			}
+
+			if (!date) {
+				const eventDate = RRule.fromString(event.date).after(new Date(), true);
+				date = eventDate ? eventDate : new Date();
```

Comment:
timezone issue

Reply:
I belive this is fixed with d5acf6712d89f7f2dfcb7e314c6083ae2dc96a36 but tbh I'm not sure.

Reply:
is this fixed on client as well?

Reply:
yes